### ISSUE TITLE
Improper connections between multiple 3d scatter viewers when selecting

### ISSUE THREAD

**Issue by PennyQ** [glue-viz/glue-vispy-viewers#105]
There are two different data sets displayed in the same Glue window using two individual 3d scatter plot viewers, while the selected region in one viewer will also be shown in another viewer, even with different XYZ attribute settings. 

![image](https://cloud.githubusercontent.com/assets/13411839/14291750/04748d7a-fb33-11e5-9762-1cd7e3856c6e.png)

**Comment by astrofrog** [comment]
The `add_subset` for scatter plots probably needs a clause like:

``` python
        if subset.data not in self._layer_artist_container:
            return
```

**Comment by PennyQ** [comment]
@astrofrog The above line is already included in the `add_subset` and doesn't work.

I guess the reason is that the `__init__.py` under the `scatter` directory is only called once, even users created two `ScattrePlots` with two different data sets. So we are actually using the same `VispyScatterViewer` for two `ScatterPlots` and that's why the `layer_artists` is the same and keep the selection 'sync' in two `ScatterPlots`. But I don't know how to solve it :(

**Comment by PennyQ** [comment]
A problem for other viewers too, one viewer selection create a subset for both viewers, as shown below:
![image](https://cloud.githubusercontent.com/assets/13411839/14753222/e1f4e9da-08a1-11e6-8bf3-eeed80855312.png)

**Comment by astrofrog** [comment]
@PennyQ - it's normal that the subset appears for all datasets - but notice that it should have a red 'x' for datasets it doesn't apply to. This wasn't working in the 3D case and I've now fixed it in #151.
